src/test/java/org/testng/JarFileUtilsTest.java

```diff
@@ -0,0 +1,113 @@
+package org.testng;
+
+import org.testng.annotations.Test;
+import org.testng.xml.IPostProcessor;
+import org.testng.xml.XmlClass;
+import org.testng.xml.XmlSuite;
+import org.testng.xml.XmlTest;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class JarFileUtilsTest {
+    private static final File jar = new File("src/test/resources/testng-tests.jar");
```

Comment:
Note for later: the jar should/could be generated by the test (for example with https://github.com/shrinkwrap/shrinkwrap)

Reply:
Added a TODO